aerogear/mobile-security-service
# fix(cover) : Fix: coveralls are count mock files

## Motivation
https://issues.jboss.org/browse/AEROGEAR-8575

## What
Fix: coveralls are count mock files

## Why
When a new mock file is generated by the lib/dep then it should not decrease the coveralls report since no make sense creates tests for these files. 

## How
- Changing the name in order to follow the mock dep README
- Adding important info in the README in order to share how the name of the mock should be. 

## Verification Steps
1. Check the % of coverage in the pkg/apps/ as the following example. 

```shell
$  go test -failfast -tags=integration -coverprofile=coverage.out -covermode=count github.com/aerogear/mobile-security-service/pkg/web/apps
ok  	github.com/aerogear/mobile-security-service/pkg/web/apps	0.058s	coverage: 54.4% of statements
```
2. Then add a new mock file in github.com/aerogear/mobile-security-service/pkg/web/apps

```shell
$ cd pkg/web/apps/

$ moq -out apps_http_handler_mock_test.go . HTTPHandler
```

3. Check that the new file was created.

```shell
$ git status
On branch AEROGEAR-8575
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	apps_http_handler_mock_test.go
```
4. Check again the % of coverage in the pkg/apps/

```shell
$  go test -failfast -tags=integration -coverprofile=coverage.out -covermode=count github.com/aerogear/mobile-security-service/pkg/web/apps
ok  	github.com/aerogear/mobile-security-service/pkg/web/apps	0.055s	coverage: 54.4% of statements
```

Expected Result: The coverage: 54.4% of statements need to be the same before created the mock file. 

## Checklist:


## Progress


## Additional Notes

<img width="763" alt="screenshot 2019-02-21 at 18 47 44" src="https://user-images.githubusercontent.com/7708031/53193569-31d29280-3609-11e9-95cc-7fee0c139bb0.png">
